### PR TITLE
Tweak library to make it usable by magnet-client

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

**Comment by wilsonpage** [comment]
@arcturus r?

The first commit also has some logic relating to throttling the event dispatch but it's a bit lost in the rename :(

File: src/main/java/org/mozilla/magnet/net/scanner/ble/ScannerBLE.java [hunk 1]
```diff
@@ -7,72 +7,68 @@
...
  */
-public class BTLEScanner extends BaseScanner {
-    private final String TAG = BTLEScanner.class.getName();
+public class ScannerBLE extends BaseScanner {
+    private final String TAG = ScannerBLE.class.getName();
     private BluetoothAdapter mBTAdapter;
     private Context mContext = null;
     private BluetoothAdapter.LeScanCallback mScanCallback = null;
     private final static String TYPE = "btle";
+    private MaxSizeHashMap<String, Long> mNotifyHistory = new MaxSizeHashMap<String, Long>(20);
```
Comment (arcturus) [inline src/main/java/org/mozilla/magnet/net/scanner/ble/ScannerBLE.java]: this declaration should be `final`, and the 20 a constant :P

**Comment by arcturus** [comment]
I cannot check MagnetScannerItem, I guess this is in the library changes right?

File: src/main/java/org/mozilla/magnet/net/scanner/MagnetScannerItem.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    public JSONObject toJSON() {
+        JSONObject json = new JSONObject();
+
+        try {
+            json.put("url", mUrl);
+            json.put("distance", mDistance);
+            json.put("type", mType);
+        } catch (JSONException e) {
+            e.printStackTrace();
```
Comment (arcturus) [inline src/main/java/org/mozilla/magnet/net/scanner/MagnetScannerItem.java:32]: We should return something here

File: src/main/java/org/mozilla/magnet/net/scanner/mdns/ScannerMDNS.java [hunk 3]
```diff
@@ -49,22 +50,18 @@ public void onDiscoveryStarted(String regType) {
...
-                    }
+
+                // not all mdns advertisements are urls
+                if (!URLUtil.isNetworkUrl(name)) {
+                    return;
                 }
+
+                MagnetScannerItem item = new MagnetScannerItem();
+                item.setUrl(name);
+                item.setType(service.getServiceType());
```
Comment (arcturus) [inline src/main/java/org/mozilla/magnet/net/scanner/mdns/ScannerMDNS.java]: Is this necessary?

Is not going to be overwritten here: https://github.com/mozilla-magnet/magnet-scanner-android/pull/5/files#diff-8a3b8412dec49dc1c0fe3b0ed1d7bdb7R45

File: src/main/java/org/mozilla/magnet/net/scanner/MagnetScannerItem.java (hunk 2 above)
Comment (wilsonpage) [inline src/main/java/org/mozilla/magnet/net/scanner/MagnetScannerItem.java:32]: Dunno. What should be returned, `json`?

**Comment by arcturus** [comment]
Looks good to me, we could bump the package version in npm.

r+!

File: src/main/java/org/mozilla/magnet/net/scanner/MagnetScannerItem.java (hunk 2 above)
Comment (arcturus) [inline src/main/java/org/mozilla/magnet/net/scanner/MagnetScannerItem.java:32]: Either we return an empty json, or we through the exception and need to handle by any client of the library (that second option looks more java), but makes the code of the clients look uglier.

**Pull request merged**